src/lib.rs

```diff
@@ -352,6 +352,93 @@ impl Filter {
     }
 }
 
+/// The possible input methods.
+pub enum InputMethod {
+    /// Read the input from the local Git history.
+    GitHistory,
+
+    /// Read the specified input file.
+    LogFile(String),
+
+    /// Read from `stdin`.
+    Stdin,
+}
+
+impl InputMethod {
+    /// Create a new instance from the given command line options.
+    pub fn parse(matches: &getopts::Matches) -> Option<Self> {
+        let mut method = 0;
+
+        if matches.opt_present("git") {
+            method |= 1
+        }
+
+        if matches.opt_present("input") {
+            method |= 2
+        }
+
+        if matches.opt_present("stdin") {
+            method |= 4
+        }
+
+        let method = method;
+
+        if method == 1 {
+            Some(Self::GitHistory)
+        } else if method == 2 {
+            Some(Self::LogFile(matches.opt_str("input").unwrap()))
```

Comment:
Question: What happens if you call the program in your branch like this:
```shell
cargo run -- --input
```
Is this properly handled by `getopts` or could this `unwrap()` here fail eventually?

Reply:
`getopts` will raise a `getopts::Fail::ArgumentMissing(argument_name)`.  This is already handled in `main.rs`, beginning in line 85.

Reply:
See https://github.com/wert007/commit-analyzer/pull/18/commits/6c26105ec9f4c5258bb84b016f4c9d39f865e1a5 for details.